lib/kubernetes-deploy/container_logs.rb

```diff
@@ -62,16 +62,23 @@ def rfc3339_timestamp(time)
 
     def deduplicate(logs)
       deduped = []
-      check_for_duplicate = true
 
-      logs.each do |line|
-        timestamp, msg = split_timestamped_line(line)
-        next if check_for_duplicate && likely_duplicate?(timestamp)
-        check_for_duplicate = false # logs are ordered, so once we've seen a new one, assume all subsequent logs are new
+      parsed_lines = logs.map { |line| split_timestamped_line(line) }
+      sorted_lines = parsed_lines.sort do |a, b|
+        if a[0].nil? || b[0].nil?
+          0
+        else
+          a[0] <=> b[0]
+        end
+      end
+
+      sorted_lines.each do |line|
+        timestamp = line[0]
+        msg = line[1]
+        next if likely_duplicate?(timestamp)
```

Comment:
> Or is it possibly true because of a previous call to this method

Exactly. We fetch logs (and then process them here) repeatedly, and since the API does not support subsecond granularity, we inevitably sometimes get the same log lines in more than one batch.